### ISSUE TITLE
Analyzer proposal: prefer `or` instead of multiple `case` that follow through

### ISSUE THREAD

**Issue by ZacharyPatten** [DotNetAnalyzers/StyleCopAnalyzers#3721]
This would probably be a very niche analyzer and thus should be a very low priority, but still, the following code:
```cs
switch (Console.ReadKey(true).Key)
{
	case ConsoleKey.D1:
	case ConsoleKey.NumPad1:
		/* ... */;
		break;
}
```
could be simplified to:
```cs
switch (Console.ReadKey(true).Key)
{
    case ConsoleKey.D1 or ConsoleKey.NumPad1:
        /* ... */;
        break;
}
```
I've seen this pattern several times when people are porting older .NET Framework code to modern .NET.

**Comment by bjornhellander** [comment]
I think the "simplified" code is **much** less readable than the original, especially for switch statements with more than two cases. I do not think that forcing users to use the second form is a good idea.

**Comment by ZacharyPatten** [comment]
Obviously, I disagree, but yeah this is very nit-picky. Nothing is stopping you from doing:
```cs
switch (Console.ReadKey(true).Key)
{
    case ConsoleKey.D1 or
         ConsoleKey.NumPad1:
        /* ... */;
        break;
}
```
if the reason you prefer the former was each condition/case being on its own line.

I think having the `or` is easier to read and it prevents code bugs like if you accidentally forget to fill in the code for one of the cases. `or` is just more clearly intentional.

**Comment by sharwell** [comment]
I also think multiple cases is more readable.
